qt-bugs@ issue : 28108
applied: no
author: John Firebaugh <jfirebaugh@kde.org>

void QDragObject::setPixmap(QPixmap pm) documentation:

    Uses a hotspot that positions the pixmap below and to the right of
    the mouse pointer. This allows the user to clearly see the point
    on the window which they are dragging the data onto.

This is a good idea. Unfortunately the code doesn't do that.

Hi John,

thanks for bringing this discrepancy to our attention; we will look into
this for a future version of Qt (we cannot change the actual behavior
within the release cycle of Qt 3.2 I'm afraid).

Regards,
Volker

--
Volker Hilsheimer, Support Manager
Trolltech AS, Waldemar Thranes gate 98, NO-0175 Oslo, Norway

Index: src/kernel/qdragobject.cpp
===================================================================
RCS file: /home/kde/qt-copy/src/kernel/qdragobject.cpp,v
retrieving revision 1.55
diff -u -3 -p -r1.55 qdragobject.cpp
--- src/kernel/qdragobject.cpp	23 Jun 2003 11:48:23 -0000	1.55
+++ src/kernel/qdragobject.cpp	2 Aug 2003 03:45:48 -0000
@@ -363,7 +363,7 @@ void QDragObject::setPixmap(QPixmap pm, 
 */
 void QDragObject::setPixmap(QPixmap pm)
 {
-    setPixmap(pm,QPoint(pm.width()/2,pm.height()/2));
+    setPixmap(pm,QPoint(-10,-10));
 }
 
 /*!
